SPHINXsys/src/shared/simbody_sphinxsys/xml_engine.h

```diff
@@ -69,100 +68,72 @@ namespace SPH
 		void addChildToElement(SimTK::Xml::Element &father_element, const std::string &child_name);
 
 		/** Add an attribute of type string to an xml element.  */
-		template <class T>
-		void setAttributeToElement(const SimTK::Xml::element_iterator &ele_ite, const std::string &attrib_name, const T &value);
-
-		void setAttributeToElement(const SimTK::Xml::element_iterator &ele_ite, const std::string &attrib_name, const int &value)
+		void setAttributeToElement(const SimTK::Xml::element_iterator &ele_ite, const std::string &attrib_name, const Real &value)
```

Comment:
I think the modification of the member function "SPH::XmlEngine::setAttributeToElement" will cause some issues.

For the last input parameter of the member function "SPH::XmlEngine::setAttributeToElement", I think several overloaded versions are missed. For example, when the data type of the last input parameter is "Vecd" or "_Ty", some error occurs like this:

![image](https://user-images.githubusercontent.com/43446342/219295537-b148b182-c4f2-4d92-8d19-518f3ae660d1.png)

Reply:
It seems that this issue during compiling on occurs for Windows.  Linux doesn't have this issue and the cases can pass the ctest. Then, I will test if I can define a 3d vector/matrix in 2D cases and also 2d vector/matrix in 3D cases in Linux.